components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/PostAuthnMissingClaimHandler.java

```diff
@@ -171,10 +177,36 @@ protected PostAuthnHandlerFlowStatus handlePostAuthenticationForMissingClaimsReq
                     }
                 }
 
+                List<LocalClaim> localClaims = getClaimMetadataManagementService().getLocalClaims(context.getTenantDomain());
+                Map<String, String> displayName = new HashMap<>();
+
+                for (Map.Entry<String, String> entry : missingClaimMap.entrySet()) {
+                    for (LocalClaim localClaim : localClaims){
+                        if (entry.getValue().equalsIgnoreCase(localClaim.getClaimURI())) {
+                            displayName.put(entry.getValue(), localClaim.getClaimProperties().get(DISPLAY_NAME_PROPERTY));
+                            break;
+                        }
+                    }
+                }
+
+                StringBuilder displayNameString = new StringBuilder();
+                StringBuilder displayNameValuesString = new StringBuilder();
+
+                for (Map.Entry<String, String> claimDisplay : displayName.entrySet()) {
+                    displayNameString.append(claimDisplay.getKey());
+                    displayNameValuesString.append(claimDisplay.getValue());
+                    displayNameString.append(",");
+                    displayNameValuesString.append(",");
+                }
+
+                String[] displayNames = {displayNameString.toString(), displayNameValuesString.toString()};
```

Comment:
I see only displayNames[1] being used, any reason to add `displayNameString.toString()` also to the array?

Reply:
Here whole set of displaynames are storing in a stringJoiner. In orderto get that set of displaynames as one string, here it used _**displayNameString.toString()**_

Reply:
i want referring to the two string method but adding both `displayNameString` and  `displayNameValuesString` to the `displayNames` array when only `displayNames[1]` is being read again